tools/coverity/misra.config

```diff
@@ -0,0 +1,26 @@
+// MISRA C-2012 Rules
+
+{
+    version : "2.0",
+    standard : "c2012",
+    title: "Coverity MISRA Configuration",
+    deviations : [
+        // Disable the following rules.
+        {
+            deviation: "Rule 2.4",
+            reason: "Allow unused tags. Some compilers warn if types are not tagged."
+        },
+        {
+            deviation: "Rule 2.5",
+            reason: "Allow unused macros. Library headers may define macros intended for the application's use, but not used by a specific file."
+        },
+        {
+            deviation: "Rule 3.1",
+            reason: "Allow nested comments. Documentation blocks contain comments for example code."
```

Comment:
Yeah there are 18 violations of this rule, it shows up because of the link to the website that is true. A more common issue is the blocks of example code we have for doxygen, a few examples:
[One](https://github.com/aws/Device-Shadow-for-AWS-IoT-embedded-sdk/blob/main/source/include/shadow.h#L768)

[Two](https://github.com/aws/Device-Shadow-for-AWS-IoT-embedded-sdk/blob/main/source/include/shadow.h#L716)

[Three](https://github.com/aws/Device-Shadow-for-AWS-IoT-embedded-sdk/blob/main/source/include/shadow.h#L57)